client-internal/src/main/java/org/terracotta/angela/client/support/junit/NodeOutputRule.java

```diff
@@ -82,17 +86,11 @@ private void onNewLine(String line) {
     // Angela log lines are standard:
     // 2020-02-26 08:19:37.894 INFO  o.t.a.e.tsa:98 - [node-1-1] AngelaMatchersxyz
     // 2020-02-26 08:19:38.640 INFO  o.t.a.e.tsa:98 - [node-1-1] xyz
-    final int start = line.indexOf(" - [node-");
-    if (start != -1) {
-      final int middle = line.indexOf('-', start + 9);
-      if (middle != -1) {
-        final int end = line.indexOf(']', middle + 1);
-        if (end != -1) {
-          final int stripeId = Integer.parseInt(line.substring(start + 9, middle));
-          final int nodeId = Integer.parseInt(line.substring(middle + 1, end));
-          getLog(stripeId, nodeId).append(line);
-        }
-      }
+    Matcher m = NODE_FINDER.matcher(line);
+    if (m.find()) {
+      int stripe = Integer.parseInt(m.group(1));
+      int node = Integer.parseInt(m.group(2));
+      getLog(stripe, node).append(line.substring(line.indexOf(m.group()) + m.group().length()));
```

Comment:
Console logs on the server still go to System.out.  System.out in now just in a single process rather than getting piped.  We need to know what System.out is coming from what server.  This is achieved with custom log config plus this class.

Reply:
I saw that and know that (I wrote this class) to cleanup angela. But I was wondering if with inline servers there would be a way (since they are in isolated classloaders) to each one have a different logging config that we could grab to directly separate the stream and avoid this "parsing".

`NodeOutputRule` works ONLY in the case of a kit started with processes when angela is pre-pending each log line with the server name (it is done in `Distribution107Controller` line #111). But in the case of inline server, there is no capture of the logs, they all go straight to stdout, and angela has no way to then capture them to prepend them by the node name. 

So my point was:
- `NodeOutputRule` is meant to be used with the way angela is capturing and rethrowing the logs by prepending the node name
- If we want to be able to use `NodeOutputRule` with inline servers, do we have a logging config per inline server that would prepend the node name exactly like angela was doing ?
- otherwise, why not simply get rid of `NodeOutputRule` for inline servers if we could capture the logging output from them ?

Reply:
This change was just to make log config changes easier to match.  The original pattern was hard to emulate in usage.  This is the logback configuration change in usage.  If you are not happy with this change, We can define a new rule closer to usage or try and modify config below to match the original pattern.
```
  <appender name="SYSOUT" class="ch.qos.logback.core.ConsoleAppender">
    <target>System.out</target>
    <encoder>
      <pattern>[${serverId}] - %d [%t] %p %c - %m%n</pattern>
    </encoder>
  </appender>
```
I'm not aware of any custom appenders that allow capture.  Maybe redirect to file and scan that but that also requires logging configuration changes and test modification.

Reply:
ok so if I understand correctly:
1. logging from core will be modified to include the node name at the start of each line so that `NodeOutputRule` can capture that in inline mode ?
2. If not in inline mode, then angela will prepend to each log line the node name again, which will duplicate what we have in the logs ?

Reply:
correct